clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestState.java

```diff
@@ -66,13 +65,19 @@ public boolean canSendRequest(final long currentTimeMs) {
             return true;
         }
 
-        if (this.lastReceivedMs == -1 ||
-                this.lastReceivedMs < this.lastSentMs) {
-            // there is an inflight request
+        if (this.lastReceivedMs == -1 || this.lastReceivedMs < this.lastSentMs) {
+            log.debug("An inflight request already exists; lastReceivedMs: {}, lastSentMs:  {}", lastReceivedMs, lastSentMs);
```

Comment:
i actually wonder if the debug would be useful due to lack of the information of what request it is.  maybe this log should stay at the request manager level?

Reply:
I added a relevant `toString()` method to the `RequestState` sub-classes that includes details about the request. The logging now includes the `toString()`-ed `RequestState` for context.